### PR TITLE
Surface build warnings in fossa analyze

### PR DESCRIPTION
**Pull request by JeffreyHuynh1**
# Overview

- When builds are sent to the [customUploadBuildHandler](https://github.com/fossas/FOSSA/blob/develop/modules/customBuildUploadHandler.ts) sometimes errors are returned under build.warnings as [pointed out by Andrew](https://teamfossa.slack.com/archives/C043EM3L96Z/p1708717893834989?thread_ts=1708712806.856329&cid=C043EM3L96Z). However, these errors aren’t being surfaced to the user in any way.

- Alex [demonstrated](https://teamfossa.slack.com/archives/CTN0VM59S/p1708718846798719?thread_ts=1708714534.565389&cid=CTN0VM59S) that these errors are present in the response 

Deserialize the warnings from the response [here ](https://github.com/fossas/fossa-cli/blob/24ed7b9bdda580c04c6bb840e92fda29eb28138c/src/Fossa/API/Types.hs#L579)

[slack thread](https://teamfossa.slack.com/archives/C043EM3L96Z/p1708718725618439?thread_ts=1708712806.856329&cid=C043EM3L96Z)

## Acceptance criteria

Surface these warnings when we do fossa analyze and include them in our debug logs

## Testing plan

fossa analyze --project-label “super long project label ….” --debug 

Screenshot 2024-02-26 at 3 16 36 PM

## References

- [ANE-1500](https://fossa.atlassian.net/browse/ANE-1500)

## Checklist

- [x] I added tests for this PR's change (or explained in the PR description why tests don't make sense).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] If this PR added docs, I added links as appropriate to the user manual's ToC in `docs/README.ms` and gave consideration to how discoverable or not my documentation is.
- [x] If this change is externally visible, I updated `Changelog.md`. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] If I made changes to `.fossa.yml` or `fossa-deps.{json.yml}`, I updated `docs/references/files/*.schema.json` AND I have updated example files used by `fossa init` command. You may also need to update these if you have added/removed new dependency type (e.g. `pip`) or analysis target type (e.g. `poetry`).
- [x] If I made changes to a subcommand's options, I updated `docs/references/subcommands/<subcommand>.md`.

[ANE-1500]: https://fossa.atlassian.net/browse/ANE-1500?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JeffreyHuynh1** [comment]
I opted not to include these warnings in our Scan Summary bc they came from uploading and not our scan process. Let me if you think differently as that will require updates to how we render to our Scan Summary file.

File: src/App/Fossa/Analyze/Upload.hs [hunk 1]
```diff
@@ -211,3 +216,24 @@ buildProjectSummary project locator projectUrl = do
       , "url" .= projectUrl
       , "id" .= renderLocator locator
       ]
+
+checkUploadResponseForWarnings ::
```
Comment (jssblck) [inline src/App/Fossa/Analyze/Upload.hs:220]: [nit] This function does more than its name implies: it _emits_ the returned warnings into the debug bundle. Consider a name like `emitBackendWarningsToBundle` instead.

**Review by jssblck — APPROVED** [review]
LGTM after changes!

File: src/App/Fossa/Analyze/Upload.hs [hunk 2]
```diff
@@ -211,3 +216,24 @@ buildProjectSummary project locator projectUrl = do
...
+  ( Has Diagnostics sig m
+  , Has Logger sig m
+  ) =>
+  UploadResponse ->
+  m ()
+checkUploadResponseForWarnings res = do
+  let maybeUploadWarnings = uploadWarnings res
+  maybe (pure ()) logUploadWarnings maybeUploadWarnings
```
[the remark this answers is not in the document]
Comment (jssblck) [reply src/App/Fossa/Analyze/Upload.hs:228]: Oh, also, make sure to do something to indicate these warnings came from the backend. The simplest way to do that is probably to wrap this function in something like `context "emit backend warnings"`.

**Pull request merged**